ibis/backends/pyspark/__init__.py

```diff
@@ -411,11 +411,18 @@ def _register_udfs(self, expr: ir.Expr) -> None:
             self._session.udf.register(f"unwrap_json_{typ.__name__}", unwrap_json(typ))
         self._session.udf.register("unwrap_json_float", unwrap_json_float)
 
+    def _in_memory_table_exists(self, name: str) -> bool:
+        sql = f"SHOW TABLES IN {self.current_database} LIKE '{name}'"
```

Comment:
Somehow `self._session.catalog.tableExists` doesn't give the same answer as the code I have here, even when passing a fully quoted `default.$memtable` (the second `dbName` parameter is deprecated and the deprecation warning suggests passing the fully qualified name.)